cuda/dot_benchmark.cu

```diff
@@ -0,0 +1,197 @@
+    constexpr std::size_t benchmark_reference{0};
+    using benchmark_info_t =
+        std::tuple<std::string, std::function<void(matrix_info, matrix_info)>,
+                   std::function<value_type()>>;
+    std::array<benchmark_info_t, benchmark_num> benchmark_info = {
+        benchmark_info_t{"DOT fp64",
+                         [&](matrix_info x_info, matrix_info y_info) {
+                             dot(device_prop, x_info, ar_data.gpu_x(), y_info,
+                                 ar_data.gpu_y(), ar_data.gpu_res());
+                         },
+                         ar_get_result},
+        benchmark_info_t{"DOT fp32",
+                         [&](matrix_info x_info, matrix_info y_info) {
+                             dot(device_prop, x_info, st_data.gpu_x(), y_info,
+                                 st_data.gpu_y(), st_data.gpu_res());
+                         },
+                         st_get_result},
+        benchmark_info_t{"DOT Acc<fp64, fp64>",
+                         [&](matrix_info x_info, matrix_info y_info) {
+                             acc_dot(device_prop, x_info, ar_data.gpu_x(),
+                                     y_info, ar_data.gpu_y(),
+                                     ar_data.gpu_res());
+                         },
+                         ar_get_result},
+        benchmark_info_t{"DOT Acc<fp64, fp32>",
+                         [&](matrix_info x_info, matrix_info y_info) {
+                             acc_dot(device_prop, x_info, st_data.gpu_x(),
+                                     y_info, st_data.gpu_y(),
+                                     ar_data.gpu_res());
+                         },
+                         ar_get_result},
+        benchmark_info_t{"DOT Acc<fp32, fp32>",
+                         [&](matrix_info x_info, matrix_info y_info) {
+                             acc_dot(device_prop, x_info, st_data.gpu_x(),
+                                     y_info, st_data.gpu_y(),
+                                     st_data.gpu_res());
+                         },
+                         st_get_result},
+        benchmark_info_t{"CUBLAS DOT fp64",
+                         [&](matrix_info x_info, matrix_info y_info) {
+                             cublas_dot(cublas_handle.get(), x_info,
+                                        ar_data.gpu_x(), y_info,
+                                        ar_data.gpu_y(), ar_data.gpu_res());
+                         },
+                         ar_get_result},
+        benchmark_info_t{"CUBLAS DOT fp32",
+                         [&](matrix_info x_info, matrix_info y_info) {
+                             cublas_dot(cublas_handle.get(), x_info,
+                                        st_data.gpu_x(), y_info,
+                                        st_data.gpu_y(), st_data.gpu_res());
+                         },
+                         st_get_result}};
+
+    std::cout << "Vector Size";
+    if (!detailed_error) {
```

Comment:
No, the opposite. Enabling the error does no longer print the runtime. This is done to prevent this benchmark to take forever, otherwise, running the detailed error would take more than 10x the amount of time of the normal benchmark.